packages/playwright-core/src/utils/isomorphic/stringUtils.ts

```diff
@@ -47,31 +47,13 @@ export function toSnakeCase(name: string): string {
   return name.replace(/([a-z0-9])([A-Z])/g, '$1_$2').replace(/([A-Z])([A-Z][a-z])/g, '$1_$2').toLowerCase();
 }
 
-export function cssEscape(s: string): string {
-  let result = '';
-  for (let i = 0; i < s.length; i++)
-    result += cssEscapeOne(s, i);
-  return result;
+export function cssEscape(selector: string): string {
+  // We are escaping it for document.querySelectorAll, not for usage in CSS file.
+  return selector.replace(/[:]/g, char => '\\' + char);
```

Comment:
Will this work for bad `.className`, e.g. `div.123`? We use `cssEscape` for class names [here](https://github.com/microsoft/playwright/blob/main/packages/injected/src/selectorGenerator.ts#L416C37-L416C46).